app/models/post.rb

```diff
@@ -1,3 +1,15 @@
 class Post < ApplicationRecord
-  belongs_to :author
+  belongs_to :user, class_name: 'User', foreign_key: 'author_id'
+  has_many :comments, dependent: :destroy
+  has_many :likes, dependent: :destroy
+
+  after_save :update_post_counter
+
+  def update_post_counter
+    user.update(posts_counter: user.posts.count)
+  end
+
```

Comment:
- [OPTIONAL] Good job by using one of the [Rails Active Callbacks (https://guides.rubyonrails.org/active_record_callbacks.html) - `after_save` then you should make the `update_post_counter` method private 🙃 🙃. Do the same for `update_like_counter` and `update_comment_counter` methods 😁 😁